controllers/configmap_controller.go

```diff
@@ -170,6 +176,60 @@ func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServ
 	return nil
 }
 
+// removeOutdatedServicesConfigMaps deletes any outdated services ConfigMaps, if all nodes have moved past that version
+func (r *ConfigMapReconciler) removeOutdatedServicesConfigMaps(ctx context.Context) error {
+	nodes := &core.NodeList{}
+	if err := r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return err
+	}
+	nodeVersions := getWindowsNodeVersions(nodes.Items)
```

Comment:
No need to loop the nodes/version twice. You can move this into `isTiedToRelevantVersion(cm.Name, nodes)`